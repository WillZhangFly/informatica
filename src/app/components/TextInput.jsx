import React from "react";

export default function TextInput({ name, id, value, handleChange }) {
  return (
    <>
      <label htmlFor={id} className="col-sm-2 col-form-label mt-3">
        {name}
      </label>
      <div className="col-sm-4">
        <input
          value={value}
          onChange={handleChange}
          type="text"
          className="form-control mt-3"
          id={id}
        />
      </div>
    </>
  );
}
