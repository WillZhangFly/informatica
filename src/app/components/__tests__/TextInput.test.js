import TextInput from "../TextInput";
import React from "react";
import { shallow } from "enzyme";

describe("TextInput component ", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <TextInput
        handleChange={() => {}}
        value="test"
        key="test"
        name="testName"
        id="testId"
        dataType="String"
      />
    );
  });
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    expect(wrapper.find("input").length).toEqual(1);
  });
});
