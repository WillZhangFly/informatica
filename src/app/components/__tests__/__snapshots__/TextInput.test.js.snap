// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component  should render 1`] = `ShallowWrapper {}`;
